docs/cluster_api.md

```diff
@@ -93,3 +80,37 @@ In more detail:
   - LastTransitionTime [string]
   - Message [string]
   - Reason [string]
+
+### MachineSetConfig
+  - `NodeType [NodeType:string]` *Enumerated node type for the machine set (compute or master)*
+  - `Infra [boolean]` *True if this machine set should contain infrastructure pods (registry, router)*
+  - `Size [int]` *How many machines should be created for this machine set*
+  - `Version [corev1.LocalObjectReference]` *Reference to ClusterVersion resource*
+  - `Hardware`
+    - `AWS` *-optional- contains AWS specific configuration*
+      - `InstanceType [string]` *EC2 instance type (ie. m4.xlarge)*
+  - `NodeLabels [map[string]string]` *Labels to apply to nodes created in this MachineSet*
+
+### ClusterVersion
+Standalone resource that encapsulates details of a named version. It
+defines general OpenShift details about a particular version such as 
+the image template and RPM version as well as cloud-specific information
+such as the AWS AMI name.
+
+A set of predefined ClusterVersion resources can be initially created in a shared 
+namespace (like `openshift`) to define the versions supported out of the box.
+Users are able to create their own in their own namespace and associate their
+clusters with that.
```

Comment:
Would it be reasonable to note that a version cannot ever be modified, and only deleted when it's no longer referenced by any cluster or machineset?

Reply:
we could... though I'm not sure there's precedent for doing that elsewhere in kube. Determining that there's no references to a resource is hard and can be racy. But I guess we could do some kind of mark and sweep. My preference would be to grab whatever we need from the currently associated version, stamp out a new machine set or whatever else we need and set the fields of that as the source of truth.

Reply:
we prevent the deletion of a PVC so long as it is in use by a pod (in 3.8)

Reply:
Is this something we would expect (someday, please not as a part of our first efforts) that this could be a ClusterVersionStream, where if we release an errata a customers ClusterVersion would immediately include it?

Is ClusterVersion required in a minimal viable product?

Reply:
> we prevent the deletion of a PVC so long as it is in use by a pod (in 3.8)

It'd be good to understand how we prevent raciness in this case. From the time that you determine that a PVC is safe to delete to the time that it's actually deleted, a pod could be created that references the PVC unless we do some kind of marking of the PVC as being in process of deletion.



> Is ClusterVersion required in a minimal viable product?

I don't think it is. But I don't think it would hurt either. I kind of like the concept of keeping all the things that work together like image tags and ami names in a single place. Do you want to leave it out of an initial implementation?